ethcore/src/engines/tendermint/mod.rs

```diff
@@ -101,6 +101,53 @@ pub struct Tendermint {
 	validators: Box<ValidatorSet>,
 }
 
+struct EpochVerifier {
+	subchain_validators: SimpleList,
+}
+
+fn combine_proofs(signal_number: BlockNumber, set_proof: &[u8]) -> Vec<u8> {
+	let mut stream = ::rlp::RlpStream::new_list(3);
+	stream
+		.append(&signal_number)
+		.append(&set_proof)
+		.out()
+}
+
+fn destructure_proofs(combined: &[u8]) -> Result<(BlockNumber, &[u8]), Error> {
+	let rlp = UntrustedRlp::new(combined);
+	Ok((
+		rlp.at(0)?.as_val()?,
+		rlp.at(1)?.data()?,
+	))
+}
+
+impl super::EpochVerifier for EpochVerifier {
+	fn verify_light(&self, header: &Header) -> Result<(), Error> {
+		let mut signatures = HashSet::new();
+		let ref header_signatures_field = header.seal().get(2).ok_or(BlockError::InvalidSeal)?;
+		for rlp in UntrustedRlp::new(header_signatures_field).iter() {
+			let signature = rlp.as_val()?; // TODO: why is `rlp` have type ()?
+			signatures.insert(signature);
```

Comment:
I am a little bit lost here. Looking at `verify_block_family`, it pattern matches on first converting the `Header` into either a `Proposal` or `VoteStep`. Then for `Proposal`s it checks if the `proposer` is an authority in the parent block and is the `proposer` of the current `proposal`, and for `VoteStep` it does the counting of signatures.

Are we to assume here that the `Header` will always correspond to a `VoteStep`?